ddl/ddl_db_test.go

```diff
@@ -277,18 +277,14 @@ func (s *testDBSuite) testAddAnonymousIndex(c *C) {
 }
 
 func (s *testDBSuite) testAddIndex(c *C) {
-	done := make(chan struct{}, 1)
-
+	done := make(chan error, 1)
 	num := defaultBatchSize + 10
 	// first add some rows
 	for i := 0; i < num; i++ {
 		s.mustExec(c, "insert into t1 values (?, ?, ?)", i, i, i)
 	}
 
-	go func() {
-		sessionExec(c, s.store, "create index c3_index on t1 (c3)")
-		done <- struct{}{}
-	}()
+	sessionExecInGoroutine(c, s.store, "create index c3_index on t1 (c3)", done)
```

Comment:
Actually you can `go sessionExecInGoroutine(...)` instead of writing `go xxx()` inner the sessionExecInGoroutine function, that can avoid an unnecessary closure.